packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -2259,3 +2260,123 @@ export class Swap extends Op {
     stack.push(b);
   }
 }
+
+// push the Nth value (0 indexed) from the top of the stack.
+// pops from stack: [...stack]
+// pushes to stack: [...stack, any (nth slot from top of stack)]
+// NOTE: dig 0 is same as dup
+export class Dig extends Op {
+  readonly line: number;
+  readonly depth: number;
+
+  /**
+   * Asserts 0 arguments are passed.
+   * @param args Expected arguments: [ depth ] // slot to duplicate
+   * @param line line number in TEAL file
+   */
+  constructor (args: string[], line: number) {
+    super();
+    this.line = line;
+    assertLen(args.length, 1, line);
+    assertOnlyDigits(args[0], line);
+
+    this.assertUint8(BigInt(args[0]), line);
+    this.depth = Number(args[0]);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, this.depth + 1, this.line);
+    const tempStack = new Stack<StackElem>(this.depth + 1); // depth = 2 means 3rd slot from top of stack
+    let copy;
+    for (let i = 0; i <= this.depth; i++) {
+      const top = stack.pop();
+      tempStack.push(top);
+      if (i === this.depth) { copy = top; }
+    }
+    while (tempStack.length()) { stack.push(tempStack.pop()); }
+    stack.push(copy as StackElem);
```

Comment:
i don't think we will need `as StackElem` here if we use the above code.

Reply:
no, we still need it.